java/src/main/scala/prometheus4cats/javasimpleclient/internal/MetricCollectionProcessor.scala

```diff
@@ -338,16 +354,11 @@ object MetricCollectionProcessor {
       _ <- Sync[F].delay(promRegistry.register(proc))
     } yield proc
 
-    def handleError(fa: F[Unit]): F[Unit] = fa.handleErrorWith { e =>
-      Logger[F].warn(e)(s"Failed to unregister a collector on shutdown.")
-    }
-
     Resource.make(acquire) { proc =>
-      handleError(Sync[F].delay(promRegistry.unregister(callbackHist))) >> handleError(
-        Sync[F].delay(
-          promRegistry.unregister(duplicateGauge)
-        )
-      ) >> handleError(Sync[F].delay(promRegistry.unregister(proc)))
+      Utils.unregister(callbackHist, promRegistry) >> Utils.unregister(
```

Comment:
Can these things throw? If so we need `guarantee` rather than `>>`. Same question for all the other occurrences, although most of them are just updating `Ref`s, which _should_ be safe

Reply:
They can, the `unregister` uses `handleErrorWith` and logs the error. It doesn't help with cancellation, but I don't think that's worth worrying about on shutdown